src/Nethermind/Nethermind.HealthChecks/NodeHealthChecks.cs

```diff
@@ -20,41 +20,51 @@
 using System.Threading;
 using System.Threading.Tasks;
 using Microsoft.Extensions.Diagnostics.HealthChecks;
+using Nethermind.Api;
 using Nethermind.Logging;
 
 namespace Nethermind.HealthChecks
 {
     public class NodeHealthCheck : IHealthCheck
     {
+        private readonly INethermindApi _api;
         private readonly INodeHealthService _nodeHealthService;
         private readonly ILogger _logger;
-        
+
         public NodeHealthCheck(
             INodeHealthService nodeHealthService,
+            INethermindApi api,
             ILogManager logManager)
         {
             _nodeHealthService = nodeHealthService ?? throw new ArgumentNullException(nameof(nodeHealthService));
+            _api = api;
             _logger = logManager.GetClassLogger();
         }
 
         public Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context, CancellationToken cancellationToken = default)
         {
             try
             {
+                if (_api.SpecProvider!.TerminalTotalDifficulty != null && !_nodeHealthService.CheckClAlive())
```

Comment:
I thought that we will add this log outside the healthchecks plugin, but I think it is fine for now :)